### ISSUE TITLE
Start POCO as Windows service with parameters

### ISSUE THREAD

**Issue by DomXC** [pocoproject/poco#2190]
#### Expected behavior
I would like to start an application (based on POCO) as Windows service. This works fine, until I use required parameters.

```
void RRCApplication::defineOptions(OptionSet& options)
{
	ServerApplication::defineOptions(options);

	options.addOption(
		Option("cfg", "f", "Load configuration data from an XML file. E.g.: /cfg=\"C:\\RRC-M\\cfg\\app.xml\"")
		.required(true)
		.repeatable(false)
		.argument("file")
		.callback(OptionCallback<RRCApplication>(this, &RRCApplication::handleConfig)));
}
```

#### Actual behavior
When starting the service, application fails on error 78. It seems, the arguments are not passed.
I attached debugger to application when starting and size of `args` is 1, so no params are passed:

\poco-1.8.0.1-all\Util\src\ServerApplication.cpp, line 179:
```
#if defined(POCO_WIN32_UTF8) && !defined(POCO_NO_WSTRING)
		std::vector<std::string> args;
		for (DWORD i = 0; i < argc; ++i) // argc == 1 (name of executable)
		{
			std::string arg;
			Poco::UnicodeConverter::toUTF8(argv[i], arg);
			args.push_back(arg);
		}
		app.init(args);
#else
```
I have registered the service as 

`sc create MySVC binpath= "C:\MySVC\MySVC.exe /cfg=\"C:\MySVC\cfg\app.xml\"" displayname= "MySVC" start= auto`

In properties of the service (in Windows service list using mmc), I can see the path:
`C:\MySVC\MySVC.exe /cfg="C:\MySVC\cfg\app.xml"`

However, the parameter **cfg** is obviously not passed. When starting application from command line, everything is OK.
The question is, **how to pass params using service**?
Thank you!

#### POCO version
1.8.0.1

#### Compiler and version
VC14

#### Operating system and version
Windows 7

#### Other relevant information
Server application as service

**Comment by aleks-f** [comment]
I remember having the same problem very long time ago, but never discovered why because having only config file was sufficient for my needs at the time and I had no time to research.

Basically, the problem can be in one of the two points:

1) poco handing of service creation/dispatch/ServiceMain
2) a windows peculiarity or bug

Looking through the code and Windows API, I don't see that poco is missing something or doing something wrong (but i've been only reading, did not run anything). It would be good if you could discover what is causing this, so we all know.

**Comment by DomXC** [comment]
OK, thank you, I hope I will find a while to investigate more...

**Comment by DomXC** [comment]
So... I have made some investigation and found the problem.

The problem is, that parameters are not passed to ``

**Comment by aleks-f** [comment]
Thanks for investigating this.

> It seems, this will be easier to implement in last version, when #if !defined(POCO_NO_WSTRING) is used only.

POCO_NO_WSTRING is not a concern on windows (and not much anywhere these days, I guess). Could you implement the fix as proposed (a special configuration file in `${application.configDir}` should be fine) and send a pull to develop branch?

**Comment by DomXC** [comment]
Hello, the fix is ready, for "develop" and for 1.8.x as well.
Fix is done using `static ArgVec _argsSvc; ` member, where args are passed to this member using `bool isService(const ArgVec& args); ` or `bool isService(int argc, wchar_t** argv);`, could this be OK? Sorry, I did not catch what you mean by special config file.

`int main(int argc, char** argv)` macro and `int ServerApplication::run(int argc, char** argv)` in develop is obsolete, am I right?
